src/indexer/index.ts

```diff
@@ -11,3 +17,44 @@ export const searchForAssets = async (tokenName: string) => {
 		console.log(e);
 	}
 };
+
+export const searchForApplication = async (application_id: number) => {
+	try {
+		const applicationInfo = await indexerClient
+			.lookupApplications(application_id)
+			.do();
+
+		const creator = applicationInfo.application.params.creator;
+		const globalState = await readAppGlobalState(creator, application_id);
+		console.log(globalState);
+		return JSON.parse(
+			JSON.stringify(applicationInfo.application.params.creator)
+		);
+	} catch (e) {
+		console.log(e);
+	}
+};
+
+export async function readAppGlobalState(
+	creator: AccountAddress,
+	appID: number
+): Promise<Map<Key, StateValue> | undefined> {
+	const accountInfoResponse = await algodClient
+		.accountInformation(creator)
+		.do();
+	for (const app of accountInfoResponse["created-apps"]) {
```

Comment:
You can also use the array filter method. No need to iterate all values.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

Reply:
actually this is a function from algobuilder. I updated it here to be used in web. Will update it after consulting with @robert-zaremba